### PR TITLE
TDP: Fix ability to build its assets; add JS unit tests

### PR DESCRIPTION
**Pull request by Scotchester**
#5818 brought the Activity Search and Building Blocks tools in from the teachers-digital-platform repository, but wasn't quite complete when it came to building the requisite assets for those tools. This PR should fix that build process.

## Additions

- `package.json` and associated Yarn files for the front-end requirements
- Adds the `tdp-analytics.js` file that was mistakenly not brought along
- Adds the JS unit tests that were not in the original PR

## Removals

- Removes `aos.css` and builds its contents into tdp's `main.less` so we don't have to check in vendor CSS
- Removes copies of JS util modules that are already in cfgov-refresh

## Changes

- Corrects locations of app-specific CSS and JS called in the templates
- Updates tdp's `main.less` to use the Design System packages that come with cfgov-refresh, rather than older Capital Framework packages
- Fixes some accidental removals of `tdp-` in CSS and JS
- Fixes typo in folder name: cfgov/unprocessed/apps/**teachers-digital-plaform** (missing the last `t`) → teachers-digital-platform
- Fixes various lint warnings

## How to test this PR

1. Pull branch
1. `yarn` – confirm that it installs dependencies from the teachers-digital-platform `package.json` file.
1. `gulp clean && gulp build`
1. Visit http://localhost:8000/practitioner-resources/youth-financial-education/journey/ and click through it
1. Visit https://www.consumerfinance.gov/practitioner-resources/youth-financial-education/teach/activities/ and run a search, select some filters, view an activity page, etc.

## Notes

- Documentation purposefully not updated until we complete the transition in the teachers-digital-platform repo
- Removes copies of utility JS modules that were already in cfgov-refresh

## Checklist

- [x] PR has an informative and human-readable title
- [x] Changes are limited to a single goal (no scope creep)
- [x] Code follows the standards laid out in the [CFPB development guidelines](https://github.com/cfpb/development)
- [ ] Project documentation has been updated

### CODE REVIEW

File: cfgov/unprocessed/apps/teachers-digital-platform/package.json [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    "cf-core": "8.0.1",
+    "cf-expandables": "8.0.1",
+    "cf-forms": "8.0.2",
+    "cf-grid": "8.0.1",
+    "cf-icons": "8.0.1",
+    "cf-layout": "8.0.1",
+    "cf-notifications": "8.0.1",
+    "cf-pagination": "8.0.1",
+    "cf-tables": "8.0.1",
+    "cf-typography": "8.0.1",
```
Comment (anselmbradford) [inline cfgov/unprocessed/apps/teachers-digital-platform/package.json:31]: Are the list of `cf-` packages here unused?

Comment (Scotchester) [reply cfgov/unprocessed/apps/teachers-digital-platform/package.json:31]: Whoops, yes, thanks. Will remove.

**Comment by willbarton** [comment]
Are the `js-lint-*` files accidental commits?

**Comment by Scotchester** [comment]
> Are the `js-lint-*` files accidental commits?

Argh, yes. I keep accidentally committing them.

**Comment by contolini** [comment]
> > Are the `js-lint-*` files accidental commits?
> 
> Argh, yes. I keep accidentally committing them.

Add `js-lint-*` to `.gitignore`?

**Comment by Scotchester** [comment]
> Add `js-lint-*` to `.gitignore`?

Nah, they're just temp files that no one but me will have.

**Comment by mistergone** [comment]
👍  This builds successfully for me!

**Comment by Scotchester** [comment]
Update: While adding the webpack-config file got the build working in Docker containers again, in testing this on the Docker stack, I notice that there are a couple aspects of the activity search that are not working (unsure if they're related):

- The "Clear all filters" button doesn't work
- Getting an error in the console with every filter change that looks to be related to the analytics code:
  ```
  Uncaught TypeError: i.getDataLayerOptions is not a function
    at s (index.14fbf8a4f37e.js:1)
    at f (index.14fbf8a4f37e.js:1)
    at HTMLDivElement.click (index.14fbf8a4f37e.js:1)
  ```

So there's more work to do here. I will press on with it tomorrow.

File: cfgov/unprocessed/apps/teachers-digital-platform/package.json [hunk 2]
```diff
@@ -0,0 +1,24 @@
+{
+  "name": "teachers-digital-platform",
+  "description": "Youth financial education activity search and building blocks journey tool",
+  "homepage": "http://localhost:8000/practitioner-resources/youth-financial-education/",
```
Comment (higs4281) [inline cfgov/unprocessed/apps/teachers-digital-platform/package.json:4]: Shouldn't the homepage point to cfgov, as other apps do?

```suggestion
  "homepage": "https://www.consumerfinance.gov/practitioner-resources/youth-financial-education/",
```

Comment (Scotchester) [reply cfgov/unprocessed/apps/teachers-digital-platform/package.json:4]: Ah, good catch!

**Comment by Scotchester** [comment]
OK! I think this is finally done. Added a clean-up commit that does the following:

- Replace duplicate utilities with their originals where possible
- Switch to imports and export in tdp-analytics.js
- Enable source maps in dev environment
- Remove TDP-specific .eslintrc

and brought in the JS unit tests that weren't included in the original PR.

Presuming all continues to test fine in the Docker stack, will merge this tomorrow morning.

**Pull request merged**